config/main.go

```diff
@@ -32,8 +44,193 @@ var (
+		Port:                   uint16(Port),
+		KeepAlive:              int32(300),
+		Timeout:                int32(300),
+		MaxConn:                int32(0),
+		ShardCronFrequency:     time.Duration(1 * time.Second),
+		MultiplexerPollTimeout: time.Duration(100 * time.Millisecond),
+		MaxClients:             int32(20000),
+		MaxMemory:              0,
+		EnvictionPolicy:        EvictionStrategy,
+		PersistanceEnabled:     true,
+	},
+	Auth: struct {
+		UserName string "mapstructure:\"username\""
+		Password string "mapstructure:\"password\""
+	}{
+		UserName: "dice",
+		Password: RequirePass,
+	},
+	Network: struct {
+		IOBufferLength    int "mapstructure:\"io-buffer-length\""
+		IOBufferLengthMAX int "mapstructure:\"io-buffer-length-max\""
+	}{
+		IOBufferLength:    IOBufferLength,
+		IOBufferLengthMAX: IOBufferLengthMAX,
+	},
+}
+
+// DiceConfig is the global configuration object for dice
+var DiceConfig *Config
+
+func SetupConfig() {
+
+	if VerifyConfigCmd() {
+		setUpViperConfig(ConfigFileLocation)
+	} else {
+		DiceConfig = &defaultConfig
+		log.Info("Using default configurations.")
+	}
+}
+
+func VerifyConfigCmd() bool {
+
+	if areBothFlagsSet() {
+		log.Error("Both -o and -c flags are set. Please use only one flag.")
+		return false
+	}
+	// case 1: 'go run main.go -o <file-path>'
+	if isValidDirPath() {
+		ConfigFileLocation = filepath.Join(CustomConfigFilePath, "dice.toml")
+		createConfigFile(ConfigFileLocation)
+		return true
+	}
+
+	return true
+
+	// case 2: 'go run main.go -c <file-path>'
+}
+
+func createConfigFile(configFilePath string) {
+
+	if _, err := os.Stat(configFilePath); err == nil {
+		log.Infof("config file already exists at %s", configFilePath)
+		return
+	}
+
+	configFile, err := os.OpenFile(configFilePath, os.O_CREATE|os.O_RDWR, 0666)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Errorf("directory does not exist at %s", CustomConfigFilePath)
+			log.Warn("starting Dice with default configurations.")
+		}
+		log.Fatalf("failed to create config file: %v", err.Error())
+	}
+	defer configFile.Close()
+	log.Infof("creating default config file at %s", configFilePath)
+
+	resp, err := http.Get("<url of config file>")
+	if err != nil {
+		log.Fatalf("failed to fetch default config file: %v", err.Error())
+	}
+	defer resp.Body.Close()
+
+	_, err = io.Copy(configFile, resp.Body)
+	if err != nil {
+		log.Fatalf("failed to write default config file: %v", err.Error())
+	}
+
+	log.Infof("config file created at %s with default configurations", configFilePath)
+}
+
+func isValidDirPath() bool {
+	if CustomConfigFilePath == constants.EmptyStr {
+		return false
+	}
+
+	info, err := os.Stat(CustomConfigFilePath)
+	if os.IsNotExist(err) || err != nil {
+		return false
+	}
+
+	if !info.IsDir() {
+		return false
+	}
+
+	return true
+}
+
+func areBothFlagsSet() bool {
+	return CustomConfigFilePath != constants.EmptyStr && ConfigFileLocation != filepath.Join(constants.DefaultConfigFilePath, "dice.toml")
```

Comment:
nit: Since not being used at multiple places can just be part of `if` statement in caller.